src/components/MapController/map-helpers.js

```diff
@@ -0,0 +1,146 @@
+/************************************************************************
+ * map-helpers.js - helper functions for MapControllers
+ * 
+ * Functions in this file fall into two general categories:
+ * 
+ *   0. Data processing helper functions, which extract information from
+ *      a MapController props object or piece of one.
+ *   
+ *   1. WMS parameter generation functions, which generate initial 
+ *      parameters for ncWMS layers of various types.
+ ************************************************************************/
+import _ from 'underscore';
+
+import { getTimeMetadata } from '../../data-services/ce-backend';
+import { getVariableOptions } from '../../core/util';
+
+/************************************************************************
+ * 0. Data processing helpers
+ ************************************************************************/
+
+function hasValidData(symbol, props) {
+  // Returns true if props contains enough information to generate a map
+  // for the primary (symbol is 'variable') or secondary ('comparand') variable.
+  var dataLocation = symbol === 'variable' ? 'meta' : 'comparandMeta';
+
+  return !_.isUndefined(props[dataLocation]) &&
+    props[dataLocation].length > 0;
+}
+
+function hasComparand(props) {
+  return this.hasValidData('comparand', props);
+}
+
+function selectedVariable(meta) {
+  // from an array of dataset meta filtered by variable, returns the active variable
+  // returns 'undefined' for empty arrays and arrays containing multiple variables.
+  var variables = _.uniq(_.pluck(meta, 'variable_id'));
+  return variables.length === 1 ? variables[0] : undefined;
+  }
+
+// predicate that detects whether a timestamp index is a 0th index:
+// either January, winter, or the first year in a file. 
+const is0thIndex = timestamp => (JSON.parse(timestamp).timeidx == 0);
+
+/********************************************************************
+ * 1. WMS parameter generating functions
+ ********************************************************************/
+
```

Comment:
See comment about promise chaining. It will change what's here if we decide to do it.